### ISSUE TITLE
Classify taking a long time

### ISSUE THREAD

**Issue by babar91** [davidsandberg/facenet#354]
When using `src/classifier.py` with my trained model, testing on only 2 images, it takes the script about 1 minute to classify (I have a decent GPU). I am trying to create a live face detection program, how can I reduce the classification time?

**Comment by MaartenBloemen** [comment]
It takes a long time because it has to load the models. To speed things up make sure you use a protobuf (.pb) file instead of checkpoint files.
During live detection and classification you load the model once after that the classification doesn't take to long.

**Comment by MyraBaba** [comment]
very stupid question :) 

I trained  a a .pkl file as described (3gig).  How can I use classifier.py to find face name from my unknown face.jpg ?

I tried such as : python src/classifier.py CLASSIFY class.pkl  tonyblair.jpg . 

always gives classifier.py: error: too few arguments

Is there any tutorial from the very beginning to explain:
training from known_faces
find faces from trained db 

txh

**Comment by MaartenBloemen** [comment]
@MyraBaba classifier.py expects a directory containing images not a single image and for classification you also need to pass the protobuf (.pb) or checkpoint files (.ckpt) from the model. This is the example of the wiki on how to use classifier.py:
`python src/classifier.py CLASSIFY ~/datasets/my_dataset/test/ ~/models/model-20170216-091149.pb ~/models/my_classifier.pkl --batch_size 1000`

**Comment by MyraBaba** [comment]
Hi,

data set : lfw_funneled
training : python src/classifier.py TRAIN  datasets/yuzler/casia_maxpy_mtcnnpy_182/   20170512-110547/20170512-110547.pb bigClass.pkl --batch_size 1000

classify : python src/classifier.py CLASSIFY  datasets/test/ 20170512-110547/20170512-110547.pb bigClass.pkl --batch_size 1000

Result : 

Number of classes: 19
Number of images: 75
Loading feature extraction model
Model filename: 20170512-110547/20170512-110547.pb
Calculating features for images
Testing classifier
Loaded classifier model from file "bigClass.pkl"
   0  Michael Schumacher: 0.002
   1  Michael Schumacher: 0.002
   2  Michael Schumacher: 0.002
   3  Michael Schumacher: 0.002
   4  Michael Schumacher: 0.002
   5  Michael Schumacher: 0.002
   6  Michael Schumacher: 0.003
   7  Halle Berry: 0.002
   8  Halle Berry: 0.002
   9  Halle Berry: 0.002
  10  Halle Berry: 0.002
  11  Halle Berry: 0.002
  12  Halle Berry: 0.002
  13  Halle Berry: 0.002
  14  Halle Berry: 0.002
  15  Halle Berry: 0.002
  16  Halle Berry: 0.002
  17  Halle Berry: 0.002
  18  Halle Berry: 0.002
  19  Halle Berry: 0.002
  20  Jennifer Capriati: 0.004
  21  Halle Berry: 0.002
  22  Halle Berry: 0.002
  23  George HW Bush: 0.002
  24  Mahmoud Abbas: 0.002
  25  Mike Weir: 0.002
  26  George W Bush: 0.003
  27  Michael Schumacher: 0.002
  28  Richard Myers: 0.002
  29  James Kelly: 0.002
  30  James Kelly: 0.002
  31  Richard Myers: 0.002
  32  James Kelly: 0.002
  33  George W Bush: 0.003
  34  Richard Myers: 0.002
  35  James Kelly: 0.002
  36  George W Bush: 0.002
  37  Mahmoud Abbas: 0.003
  38  Mahmoud Abbas: 0.002
  39  Mahmoud Abbas: 0.003
  40  Michael Schumacher: 0.002
  41  Michael Schumacher: 0.001
  42  Michael Schumacher: 0.003
  43  Michael Schumacher: 0.003
  44  Michael Schumacher: 0.003
  45  Michael Schumacher: 0.003
  46  Michael Schumacher: 0.003
  47  Michael Schumacher: 0.003
  48  Michael Schumacher: 0.003
  49  Michael Schumacher: 0.003
  50  Michael Schumacher: 0.003
  51  Michael Schumacher: 0.003
  52  Michael Schumacher: 0.003
  53  Michael Schumacher: 0.003
  54  Michael Schumacher: 0.003
  55  Michael Schumacher: 0.003
  56  Michael Schumacher: 0.003
  57  Michael Schumacher: 0.003
  58  Michael Schumacher: 0.003
  59  Michael Schumacher: 0.003
  60  Mahmoud Abbas: 0.001
  61  Michael Schumacher: 0.002
  62  Gerhard Schroeder: 0.002
  63  Michael Schumacher: 0.002
  64  Michael Schumacher: 0.002
  65  Michael Schumacher: 0.002
  66  Mahmoud Abbas: 0.002
  67  Mahmoud Abbas: 0.002
  68  George W Bush: 0.002
  69  Michael Schumacher: 0.002
  70  Nancy Pelosi: 0.001
  71  Arnold Schwarzenegger: 0.003
  72  Arnold Schwarzenegger: 0.002
  73  Arnold Schwarzenegger: 0.003
  74  Arnold Schwarzenegger: 0.003

Accuracy: 0.000

where is the missing point ?  accuracy too low and also lot of miss detection ?

Thx.

**Comment by MaartenBloemen** [comment]
Hmm that's weird.
Did you use the entire Casia dataset to train the pkl classifier?
Also is your protobuf (.pb) a pre trained model or one that you trained yourself?

**Comment by MyraBaba** [comment]
YEs I used the set with more than five thousand people (13.000+ pics) and pick a test inside from it ie 20 people .  So this is the result.

I used pretrained model which dowloaded from the site.  20170512-110547.pb

**Comment by MaartenBloemen** [comment]
My suggestion is that you try to train a classifier with 10 celebrities (or people you have a lot of pictures from) and for each person get 20-30 photos. Then get some random photos of the persons in your set and some of persons not in your set and see if the results are better.

**Comment by davidsandberg** [comment]
Closing this for now. Reopen if needed.

**Comment by Zumbalamambo** [comment]
@davidsandberg  how do I do live recognition?
